### DISCUSSION TITLE
Params overload for `Contain` and `NotContain` in `GenericCollectionAssertions`

### DISCUSSION THREAD

**Discussion by mu88** [fluentassertions/fluentassertions#1699] [Ideas]
Recently I had to write a couple of assertions which looked similar to this:
```cs
// Arrange
var car1 = new Car();
var car2 = new Car();
var car3 = new Car();
var car4 = new Car();
var cars = new List<Car> { car1, car2, car3, car4 };
var sut = new JunkYard();

// Act
sut.Scrap(cars, car2, car3); // removes car2 and car3 from cars

// Assert
cars.Should().Contain(new [] { car1, car4 });
cars.Should().NotContain(new [] { car2, car3 });
```

It would be great if I could write the `Assert` block like this:
```
cars.Should().Contain(car1, car4);
cars.Should().NotContain(car2, car3);
```

For this I'd need two overloads within `GenericCollectionsAssertions`:
- `public AndConstraint<TAssertions> Contain(params T[] expected)`
- `public AndConstraint<TAssertions> NotContain(params T[] unexpected)`

What do you think about this? If you like the idea, I'd love to provide a PR.

**Comment by jnyrup** [comment]
In 6.0.0 we removed _some_ `params` overloads as they can bring in ambiguity.
I thought we had removed more, but it seems we kept `[Not]ContainInOrder(params T[])` and `Equal(params T[])` for `GenericCollectionAssertions`.

If `Contain(params T[] expected)` is added, the code below will not check if `subject` contains an element `1`, but if it contains an element `new[] { 1 }`.

```cs
var subject = new List<object>();
subject.Should().Contain(new[] { 1 });
```

Related material:
https://fluentassertions.com/upgradingtov6#collections 
The attached issues to #1529

**Reply by mu88** [reply]
Ah okay, I understand. That's sad but I'll hopefully get used to it 😉 From my perspective we can close this discussion

**Reply by jnyrup** [reply]
For your own project, you can of course add an extension method to `GenericCollectionAssertions` to get the params overload.

**Reply by mu88** [reply]
That's what I do at the moment. But since I faced the issue now a second time, I thought about providing it to the core. But then I'll think about my own NuGet package.
